tests/unit_tests/feature/testgetfeaturelistscheduler.cpp

```diff
@@ -0,0 +1,160 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "testgetfeaturelistscheduler.h"
+
+#include <QTest>
+
+#include "feature/getfeaturelistscheduler.h"
+#include "feature/taskgetfeaturelist.h"
+#include "settingsholder.h"
+#include "task.h"
+#include "taskscheduler.h"
+
+void TestGetFeatureListScheduler::testTaskIsScheduledOnStart() {
+  SettingsHolder settingsHolder;
+
+  // Stop the TaskScheduler so we can inpect the tasks scheduled.
+  TaskScheduler::stop();
+
+  auto initialListOfTasks = TaskScheduler::tasks();
+
+  GetFeatureListScheduler worker;
+  worker.start(1 * 1000 * 60 * 60);  // 1hr
+
+  auto updatedListOfTasks = TaskScheduler::tasks();
+
+  QCOMPARE(initialListOfTasks.count() + 1, updatedListOfTasks.count());
+
+  // Check that the latest task is TaskGetFeatureList, just in case.
+  const TaskGetFeatureList* task =
+      dynamic_cast<const TaskGetFeatureList*>(updatedListOfTasks.takeLast());
+  // If it is not a nullptr, the cast was succesfull.
+  QVERIFY(task);
+
+  // Remove the TaskScheduler singleton so that state doesn't leak to other
+  // tests.
+  TaskScheduler::reset();
+}
+
+void TestGetFeatureListScheduler::testTaskIsScheduledPeriodically() {
+  SettingsHolder settingsHolder;
+
+  // Stop the TaskScheduler so we can inpect the tasks scheduled.
+  TaskScheduler::stop();
+
+  auto initialListOfTasks = TaskScheduler::tasks();
+
+  GetFeatureListScheduler worker;
+  worker.start(200);  // 200ms
+
+  // Sleep for 250 milliseconds,
+  // give the worker time to schedule the first periodic task.
+  QTest::qWait(250);
+
+  auto updatedListOfTasks = TaskScheduler::tasks();
+
+  // We are expecting two tasks, the task scheduled on start + 1.
+  QCOMPARE(initialListOfTasks.count() + 2, updatedListOfTasks.count());
+
+  // Check that the tasks are TaskGetFeatureList, just in case.
+  // Check that the tasks are TaskGetFeatureList, just in case.
+  for (int i = 0; i < 2; ++i) {
+    const TaskGetFeatureList* task =
+        dynamic_cast<const TaskGetFeatureList*>(updatedListOfTasks.takeLast());
+    // If it is not a nullptr, the cast was succesfull.
+    QVERIFY(task);
+  }
+
+  // Remove the TaskScheduler singleton so that state doesn't leak to other
+  // tests.
+  TaskScheduler::reset();
+}
+
+void TestGetFeatureListScheduler::testTaskIsScheduledOnTokenChange() {
+  SettingsHolder settingsHolder;
+
+  // Stop the TaskScheduler so we can inpect the tasks scheduled.
+  TaskScheduler::stop();
+
+  auto initialListOfTasks = TaskScheduler::tasks();
+
+  GetFeatureListScheduler worker;
+  worker.start(1 * 1000 * 60 * 60);  // 1hr
+
+  settingsHolder.setToken("aToken");
+  settingsHolder.setToken("anotherToken");
+  settingsHolder.removeToken();
```

Comment:
Should tests restore the original token, so it doesn't affect the machine that the test is running on?

Reply:
Same resply as https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8956#discussion_r1466512725